packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -213,21 +213,28 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
     // If a channel is proposed, join it
     this.paymentChannelClient.onChannelProposed(async (channelState: ChannelState) => {
       await this.paymentChannelClient.joinChannel(channelState.channelId);
+      log(`Joined channel ${channelState.channelId}`);
     });
 
     this.paymentChannelClient.onChannelUpdated(async (channelState: ChannelState) => {
+      if (channelState.channelId !== wire.paidStreamingExtension.pseChannelId) {
+        // filter to updates for the channel on this wire
+        return;
+      }
       log(`State received with turnNum ${channelState.turnNum}`);
-      if (
-        (channelState.channelId === wire.paidStreamingExtension.pseChannelId && // must narrow to *this* wire
-          this.paymentChannelClient.isPaymentToMe(channelState)) ||
-        Number(channelState.turnNum) === 3
-        // returns true for the second postFS if I am the beneficiary
-        // (I need to countersign this state in order for the first payment to be sent)
-      ) {
+      if (this.paymentChannelClient.shouldSendSpacerState(channelState)) {
+        // send "spacer" state
+        await this.paymentChannelClient.acceptChannelUpdate(
+          this.paymentChannelClient.channelCache[channelState.channelId]
+        );
+        wire.paidStreamingExtension.stop(); // prompt peer for a payment
+        return;
```

Comment:
Generally we shouldn't exit function like this. We only have another if below. use an `else if` is cleaner.

I point it out this because when people reading the code, it's easy to miss the a block that suddenly exits and jumps somewhere, like a `goto` statement. when that happens, people would try adding a statement down below and expect it to execute. but look and behold, it never reaches there. It's anther debugging session. So we should usually `return` at the end.

Reply:
079f1f2